lib/internal/modules/cjs/loader.js

```diff
@@ -236,15 +236,24 @@ ObjectDefineProperty(Module, 'wrapper', {
 });
 
 function getModuleParent() {
+  const parent = moduleParentCache.get(this);
+  if (!process.noDeprecation && !getModuleParent.warned &&
+      (parent == null || pendingDeprecation)) {
+    process.emitWarning(
+      'module.parent is deprecated due to accuracy issues. Please use ' +
+        'require.main to find program entry point instead.',
+      'DeprecationWarning',
+      'DEP0143'
+    );
+    getModuleParent.warned = true;
+  }
+  if (parent === undefined) {
+    return new (class UnknownModule {})();
```

Comment:
why this new class instance instead of keeping the return value the same?

Reply:
It should probably at least be cached if it’s truly necessary, otherwise it breaks the expectation that `module.parent === module.parent`.